### PR TITLE
Updates tips description field on Rewards page.

### PR DESCRIPTION
**Pull request by mkarolin**
Fixes brave/brave-browser#20827

## Submitter Checklist:

- [ ] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [ ] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

File: components/brave_rewards/resources/android_page/components/tipsBox.tsx [hunk 1]
```diff
@@ -79,6 +79,14 @@ class TipBox extends React.Component<Props, State> {
     })
   }
 
+  getDescriptionChild = () => {
```
Comment (emerick) [inline components/brave_rewards/resources/android_page/components/tipsBox.tsx:82]: Nit: I think `getDescription` would be fine here.

**Review by emerick — APPROVED** [review]
LGTM, just had one nit

**Review by szilardszaloki — APPROVED** [review]
Just a nit: not sure if @Miyayes meant "[Learn more about tipping.](https://support.brave.com/hc/en-us/articles/360021123971-How-do-I-tip-websites-and-Content-Creators-in-Brave-Rewards-)", or "[Learn more](https://support.brave.com/hc/en-us/articles/360021123971-How-do-I-tip-websites-and-Content-Creators-in-Brave-Rewards-) about tipping.".

**Review by zenparsing — APPROVED** [review]

File: components/brave_rewards/resources/android_page/components/tipsBox.tsx (hunk 1 above)
Comment (mkarolin) [reply components/brave_rewards/resources/android_page/components/tipsBox.tsx]: Changed in 6231508161154fd72b57caa05f0fa40bf641f429

File: components/brave_rewards/resources/ui/components/box/index.tsx [hunk 2]
```diff
@@ -30,7 +30,7 @@ export type Type = 'ads' | 'contribute' | 'donation'
 export interface Props {
   title: string
   id?: string
-  description?: string
+  description?: string | React.ReactNode
```
[the remark this answers is not in the document]
Comment (mkarolin) [reply components/brave_rewards/resources/ui/components/box/index.tsx]: Fixed in 6231508161154fd72b57caa05f0fa40bf641f429

File: components/brave_rewards/resources/page/components/tipsBox.tsx [hunk 3]
```diff
@@ -157,6 +157,14 @@ class TipBox extends React.Component<Props, State> {
     )
   }
 
+  getDescriptionChild = () => {
+    return (
+      <div>
+        {getLocale('donationDesc')} <a href = {'https://support.brave.com/hc/en-us/articles/360021123971-How-do-I-tip-websites-and-Content-Creators-in-Brave-Rewards-'} target = {'_blank'}>{getLocale('donationDescLearnMore')}</a>
```
[the remark this answers is not in the document]
Comment (mkarolin) [reply components/brave_rewards/resources/page/components/tipsBox.tsx]: Fixed in 6231508161154fd72b57caa05f0fa40bf641f429

**Comment by mkarolin** [comment]
@szilardszaloki I did ask that in the issue here https://github.com/brave/brave-browser/issues/20827#issuecomment-1035553978 and @Miyayes picked the option with the whole sentence being a link.

**Comment by mkarolin** [comment]
CI passed. Squashed fixup commit. Merging.

**Comment by kjozwiak** [comment]
Verification **PASSED** on `Win 11 x64` using the following build:

```
Brave | 1.37.50 Chromium: 98.0.4758.87 (Official Build) nightly (64-bit)
-- | --
Revision | e4cd00f135fb4d8edc64c8aa6ecbe7cc79ebb3b2-refs/branch-heads/4758@{#1002}
OS | Windows 11 Version 21H2 (Build 22000.493)
```

As per https://github.com/brave/brave-browser/issues/20827#issuecomment-1035553978, ensured that the new text was being used under `brave://rewards`:

**`Old`** | **`New`**
----------|----------
old | new

Ensured that clicking on `Learn more about tipping.` opens https://support.brave.com/hc/en-us/articles/360021123971-How-do-I-tip-websites-and-Content-Creators-in-Brave-Rewards- as per https://github.com/brave/brave-browser/issues/20827#issue-1120939075.

---------------

Verification **PASSED** on `Pixel 6` running `Android 12` using `1.37.50 Chromium: 98.0.4758.87`

As per https://github.com/brave/brave-browser/issues/20827#issuecomment-1035553978, ensured that the new text was being used under `brave://rewards`:

**`Old`** | **`New`**
----------|----------
image | image
image | image

Ensured that clicking on `Learn more about tipping.` opens https://support.brave.com/hc/en-us/articles/360021123971-How-do-I-tip-websites-and-Content-Creators-in-Brave-Rewards- as per https://github.com/brave/brave-browser/issues/20827#issue-1120939075.

**Pull request merged**